### PR TITLE
fix(table): targeted control-row sort button margining

### LINKED ISSUE #7220: Bug - Table - misalignment when only some headers are sortable
**Describe the problem**
This is an issue with 
- [ ] Patternfly 5  
- [x] Patternfly 6  
- [ ] other

Please give a clear and concise description of the problem. Which components are affected?
Table with sortable headers

**How do you reproduce the problem?**
Provide steps to reproduce. A [codepen](http://codepen.io) or [codesandbox](https://codesandbox.io/) demonstrating the problem is appreciated.

Go to the [React nested expandable table example](https://www.patternfly.org/components/table/#expandable-with-nested-table) and make only one of the header cells sortable. You can just pass `sort={{columnIndex: 0}}` to one of the `<Th` components that has a visible text label, do this for both the outermost table and the nested table.

Notice that the text for the sortable header button is misaligned compared to the NON-sortable header cells (see screenshot below):

Image

**Expected behavior**
A clear and concise description of the expected behavior.
All cells would be veritcally aligned regardless of the amount of sortable header cells

**Is this issue blocking you?**
List the workaround if there is one.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**What is your environment?**

**What is your product and what release date are you targeting?**
This was found during updates being made to RHOAI for v6 migration.

**Any other information?**

### PR DESCRIPTION
**Pull request by mattnolting**
closes #7220 

TLDR: Issue is being caused by table sort vertical alignment (should be `vertical-align: top`) in tbody and negative margining. 

[Backstop report](https://drive.google.com/file/d/1vVn1BdRJeiG6sThWNFXL42A-GS9PCUNj/view?usp=sharing)

Because tables may contain elements with varying `line-height`/`font-size`, all `th/td` cells, whether in `thead` or `tbody` require a consistent and explicit `line-height` to properly align vertically.

### CODE REVIEW

File: src/patternfly/components/Table/table.scss [hunk 1]
```diff
@@ -1054,6 +1049,15 @@
 .#{$table}__tbody {
   vertical-align: top;
 
+  .#{$table}__sort {
+    vertical-align: top;
+  }
```
Comment (mcoker) [inline src/patternfly/components/Table/table.scss:1054]: If the headers vary in height, this is going to make the sortable ones top aligned

Screenshot 2024-12-03 at 2 41 59 PM

**Review by sg00dwin — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

File: src/patternfly/components/Table/table.scss [hunk 2]
```diff
@@ -340,11 +343,19 @@
...
+        margin-inline-start: calc((var(--#{$table}--cell--first-last-child--PaddingInline) - var(--#{$table}--cell--PaddingInlineStart)) * -1);
+      }
     }
 
     // Last child padding right
     &:last-child {
       padding-inline-end: var(--#{$table}--cell--first-last-child--PaddingInline);
+
+      .#{$table}__button {
+        margin-inline-end: calc((var(--#{$table}--cell--first-last-child--PaddingInline) - var(--#{$table}--cell--PaddingInlineend)) * -1);
```
Comment (sg00dwin) [inline src/patternfly/components/Table/table.scss:357]:
```suggestion
        margin-inline-end: calc((var(--#{$table}--cell--first-last-child--PaddingInline) - var(--#{$table}--cell--PaddingInlineEnd)) * -1);
```

**Comment by sg00dwin** [comment]
@mattnolting I noticed that the table `th` sortable button text and the nested table first column is no longer left aligned with other table elements.
https://patternfly-pr-7244.surge.sh/components/table/html/compound-expansion-example/
https://www.patternfly.org/components/table/html/compound-expansion-example/

Screenshot 2024-12-04 at 12 52 36 PM

File: src/patternfly/components/Table/table.scss [hunk 3]
```diff
@@ -78,8 +79,6 @@
   --#{$table}--cell--hidden-visible--Display: table-cell;
 
   // * Table toggle
-  --#{$table}__toggle--PaddingBlockStart: var(--pf-t--global--spacer--sm);
```
Comment (mcoker) [inline src/patternfly/components/Table/table.scss:81]: Not sure if you meant to remove this? It's breaking to remove it, and it looks like it's still used here https://github.com/patternfly/patternfly/blob/4ee99f15fbb5760024916e185cda063f19cdda6f/src/patternfly/components/Table/table.scss#L741

File: src/patternfly/components/Table/table.scss [hunk 4]
```diff
@@ -78,8 +79,6 @@
...
-  --#{$table}__toggle--PaddingBlockEnd: var(--pf-t--global--spacer--sm);
```
Comment (mcoker) [inline src/patternfly/components/Table/table.scss:82]: Similar issue with this one - it's still used here https://github.com/patternfly/patternfly/blob/4ee99f15fbb5760024916e185cda063f19cdda6f/src/patternfly/components/Table/table.scss#L741

And then redefined here https://github.com/patternfly/patternfly/blob/4ee99f15fbb5760024916e185cda063f19cdda6f/src/patternfly/components/Table/table.scss#L1057

File: src/patternfly/components/Table/table.scss [hunk 5]
```diff
@@ -4,6 +4,7 @@
 // TODO: update grouping comments to // * Table {element}
 
 @include pf-root($table) {
+  --#{$table}--LineHeight: calc(var(--pf-t--global--font--line-height--body) * var(--pf-t--global--font--size--body--default));
```
Comment (mcoker) [inline src/patternfly/components/Table/table.scss:7]: Since this is applied to cells, any text in a cell that 1) is not `font--size--body--default` and 2) inherits `line-height` will have the wrong line-height. Our empty state demo shows this - https://patternfly-pr-7244.surge.sh/components/table/html-demos/empty-state/

`empty-state__body` font size is 16px and the line-height is 21px now (14px * 1.5) instead of 24px.

**Comment by mcoker** [comment]
Closing this PR - #7220 was closed via #7246
